plugin/hooks.go

```diff
@@ -166,6 +167,15 @@ type Hooks interface {
 	// Minimum server version: 5.2
 	MessageHasBeenUpdated(c *Context, newPost, oldPost *model.Post)
 
+	// MessagesWillBeConsumed is invoked when a message is requested by a client before it is returned
+	// to the client
+	//
+	// Note that this method will be called for posts created by plugins, including the plugin that
+	// created the post.
+	//
+	// Minimum server version: 8.1
```

Comment:
This might need to land in September's release, vs. the upcoming August, as we've already cut the release branch. I think we're tentatively labelling that 9.0 (I know, v8 was short!) to reflect some cadence changes.

Reply:
@qryptdev This one's important to change before merging, indeed.

Reply:
I'll put 9.0, but let us know if this will need to change.